hax/hax/motr/hax.c

```diff
@@ -541,7 +541,21 @@ static void msg_is_delivered_cb(struct m0_halon_interface *hi,
 static void msg_is_not_delivered_cb(struct m0_halon_interface *hi,
 				    struct m0_ha_link *hl, uint64_t tag)
 {
-	M0_LOG(M0_DEBUG, "noop");
+	struct m0_fid *proc_fid = &hl->hln_conn_cfg.hlcc_rpc_service_fid;
+
+	PyGILState_STATE gstate;
+	gstate = PyGILState_Ensure();
+
+	/* Notify hax about delivery failure. */
+	M0_LOG(M0_DEBUG, "msg not delivered, tag=%"PRIu64,
+		hl->hln_tag_broadcast_delivery);
+
+	PyObject *py_fid = toFid(proc_fid);
+	PyObject_CallMethod(hc0->hc_handler, "_msg_delivered_cb", "(OsKK)",
```

Comment:
I think it makes sense (at least) to add a bool parameter saying whether it is delivered (success story) or not delivered. I understand that at this point it doesn't make much sense since the only use case when delivery callback is used (at the moment) is DeliveryHerald mechanism support. But I would still prefer to keep the intentions and semantics clear. If it is not delivered, let's not hide the truth - it wasn't delivered.

Reply:
I think then let's have a corresponding callback function _msg_not_delivered_cb() to be more explicit.

Reply:
Sounds good to me.

Reply:
Done.